### PR TITLE
Add plugin data to be exported as personal data

### PR DESCRIPTION
**Pull request by mariovalney**
Add plugin data to be exported throught **coauthors_guest_authors_exported_extra_data** filter.

**Dependece:**

https://github.com/Automattic/Co-Authors-Plus/pull/528

### CODE REVIEW

File: co-authors-plus-social-pack.php [hunk 1]
```diff
@@ -40,18 +40,20 @@ function __construct(){
...
+		add_filter( 'jetpack_sharing_twitter_via',				   array( $this, 'filter_jetpack_sharing_twitter_via' ),	 10, 2 );
+		add_filter( 'jetpack_sharing_twitter_related',			   array( $this, 'filter_jetpack_sharing_twitter_related' ), 10, 2 );
 
-		add_filter( 'coauthors_guest_author_fields', 	array( $this, 'filter_coauthors_guest_author_fields' ), 	10, 2 );
+		add_filter( 'coauthors_guest_author_fields',			   array( $this, 'filter_coauthors_guest_author_fields' ),	 10, 2 );
 
-		add_action( 'add_meta_boxes', 					array( $this, 'action_add_meta_boxes' ), 					20, 2 );
+		add_action( 'add_meta_boxes',							   array( $this, 'action_add_meta_boxes' ),					 20, 2 );
+
+		add_filter( 'coauthors_guest_author_personal_export_data', array( $this, 'filter_export_extra_data' ),				 10, 2 );
```
Comment (mjangda) [inline co-authors-plus-social-pack.php]: Per the latest feedback, this needs a small update to `coauthors_guest_author_personal_export_extra_data` (https://github.com/Automattic/Co-Authors-Plus/pull/528#discussion_r194557539)

File: co-authors-plus-social-pack.php [hunk 2]
```diff
@@ -197,6 +199,46 @@ public function filter_coauthors_guest_author_fields( $fields_to_return, $groups
...
+
+			$key = $this->coauthors_plus->guest_authors->get_post_meta_key( $field['key'] );
+			$value = get_post_meta( $author_id, $key, true );
+
+			if ( empty( $value ) ) {
+				continue;
+			}
+
+			$data[] = array(
+				'name'	=> __( 'Guest Author', 'co-authors-plus' ) . ' ' . $field['label'],
```
Comment (mjangda) [inline co-authors-plus-social-pack.php]: We can remove the `Guest Author` prefix here.

File: co-authors-plus-social-pack.php [hunk 3]
```diff
@@ -197,6 +199,46 @@ public function filter_coauthors_guest_author_fields( $fields_to_return, $groups
...
+	 * Hook into Co Authors Plus's filter coauthors_guest_author_personal_export_data to extra fields
+	 * to exported personal data
+	 *
+	 * @param  array $data      Data to be exported
+	 * @param  array $author_id The guest author ID
+	 * @return array            The extra data to be exported
+	 */
+	function filter_export_extra_data( $data, $author_id ) {
+		$fields = $this->coauthors_plus->guest_authors->get_guest_author_fields( 'social' );
+		$ignore_fields = array(
```
Comment (mjangda) [inline co-authors-plus-social-pack.php]: Why are we ignoring these fields? We should add an inline comment with the reason.

Comment (mariovalney) [reply co-authors-plus-social-pack.php]: Because they aren't personal data fields. Comment added.

File: co-authors-plus-social-pack.php (hunk 2 above)
Comment (mariovalney) [reply co-authors-plus-social-pack.php]: Done

**Comment by mariovalney** [comment]
Thanks for your support @mjangda

**Pull request merged**